tests/clvm/test_generate_curry_and_treehash.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any
+
+import blspy
+
+from chia.types.blockchain_format.program import Program
+from chia.wallet.puzzles import p2_delegated_puzzle_or_hidden_puzzle  # import (puzzle_for_pk, puzzle_hash_for_pk, MOD)
+from chia.wallet.util.generate_curry_and_treehash_f import generate_curry_and_treehash_f
+
+
+def test_generate_curry_and_treehash_f() -> None:
+    arbitrary_mod = p2_delegated_puzzle_or_hidden_puzzle.MOD
+    # we don't really care what `MOD` is. We just need some code
+
+    f = generate_curry_and_treehash_f(arbitrary_mod.get_tree_hash())
+
+    for v in range(500):
+        args = [v, v * v, v * v * v]
+        # we don't really care about the arguments either
+        puzzle = arbitrary_mod.curry(*args)
+        puzzle_hash_via_curry = puzzle.get_tree_hash()
+        hashed_args = [Program.to(_).get_tree_hash() for _ in args]
+        puzzle_hash_via_f = f(*hashed_args)
+        assert puzzle_hash_via_curry == puzzle_hash_via_f
+
+
+def mod_api_check(module: Any) -> None:
```

Comment:
would it work to make this `object` instead of `Any`?
It looks like a WalletProtocol` would come in handy here

Reply:
At first I thought this function would be property of a module, but it turns out to be property of `Wallet` subclasses (not real subclasses, just duck-typing subclasses), and they are pretty hard to create in a vacuum, so this test either has to use a `mock` or import a lot of junk just to set up.